### PR TITLE
#159965437 Implements Sign Up for Authors Haven.

### PR DESCRIPTION
**Pull request by Mutuba**
## What does this PR do?
```
Implements sign up page for Authors Haven.
```

## Description of Task to be completed?
```
Users should be able to create an account by passing appropriate parameters.

```

## How should this be manually tested?

```
Clone the repo. cd into ah-leagueoflegends-frontend. npm install. Go to the browser. Click on Getting Started button. A register pops up. Enter your details and click on register.

```
## Any background context you want to provide?

```
Authors Haven did not have front end sign up functionality.

```

## What are the relevant pivotal tracker stories?

[#159965437](https://www.pivotaltracker.com/story/show/159965437)

## Screenshots if available to support PR.

screen shot 2018-10-10 at 17 55 08

## Questions:

```
Can it be done better?

```

### CODE REVIEW

File: src/styles/styles.scss [hunk 1]
```diff
@@ -1,2 +1,36 @@
-@import "./scss/base/base"
+@import "./scss/base/base";
 
+#modal2 {
```
[the remark this answers is not in the document]
Comment (Mutuba) [reply src/styles/styles.scss]: A component needs an id as selector in order to render.

[the remark this answers is not in the document]
Comment (Mutuba) [reply src/styles/styles.scss:5]: A modal needs an id in order to be selected so as to be rendered.

File: src/components/Authentication/Signup/Register.js [hunk 2]
```diff
@@ -0,0 +1,150 @@
+// Register.js
+
+import React, { Component } from 'react';
```
[the remark this answers is not in the document]
Comment (Mutuba) [reply src/components/Authentication/Signup/Register.js:3]: Component has been used during creation of component(Register)

File: src/styles/scss/components/_articles.scss [hunk 3]
```diff
@@ -10,10 +15,10 @@
 }
 
 .article-aside {
-  float: right;
-
-  margin: 20px 2%;
-  width: 35%;
+  float: left;
+  margin-left: 45px !important;
```
[the remark this answers is not in the document]
Comment (Mutuba) [reply src/styles/scss/components/_articles.scss:19]: Needed that in order to style the aside articles.

File: src/styles/scss/base/_settings.scss [hunk 4]
```diff
@@ -1 +1,3 @@
 $color-gray: #aaa;
+$color-red: #ff0000;
+$color-green: #008080;
```
[the remark this answers is not in the document]
Comment (Mutuba) [reply src/styles/scss/base/_settings.scss]: I have put trailing new line

File: src/components/Authentication/Signup/store.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+import { createStore, applyMiddleware, compose } from 'redux';
+import thunk from 'redux-thunk';
+import rootReducer from './reducers';
+
+const inititalState = {};
+
+const store = createStore(rootReducer, inititalState, compose(applyMiddleware(thunk)));
+
+export default store;
```
Comment (kevintumbo) [inline src/components/Authentication/Signup/store.js]: Do not create multiple stores in one project. Use the store in the stores folder instead

File: src/containers/Articles/Read/index.js [hunk 6]
```diff
@@ -5,7 +5,7 @@ import PropTypes from 'prop-types';
 
 import '../../../styles/styles.css';
 import getArticles from './actions';
-import ArticleHeader from '../../../components/Header';
+// import ArticleHeader from '../../../components/Header';
```
Comment (kevintumbo) [inline src/containers/Articles/Read/index.js]: remove commented code

File: src/index.js [hunk 7]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
-import './index.css';
+import './styles/styles.scss';
```
[the remark this answers is not in the document]
Comment (kevintumbo) [reply src/index.js]: shouldn't this be .css?

File: src/styles/styles.scss [hunk 8]
```diff
@@ -1,2 +1,36 @@
-@import "./scss/base/base"
+@import "./scss/base/base";
 
+.modal1 {
+  text-align: center;
```
[the remark this answers is not in the document]
Comment (kevintumbo) [reply src/styles/styles.scss:7]: this can be done

File: src/styles/styles.scss [hunk 9]
```diff
@@ -1,2 +1,36 @@
...
+  width: 70%;
+  background-color: rgb(219, 235, 235)
```
[the remark this answers is not in the document]
Comment (kevintumbo) [reply src/styles/styles.scss]: this too.

File: src/styles/styles.scss [hunk 10]
```diff
@@ -1,2 +1,36 @@
...
+}
+
+.modal-content p {
+  margin: 1% 25%;
```
[the remark this answers is not in the document]
Comment (kevintumbo) [reply src/styles/styles.scss]: this too.

File: src/components/Authentication/Signup/Home.js [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+//                 </ul>
+//               </div>
+//             </div>
+//           </nav>
+//         </div>
+//         <Register />
+//       </div>
+//     );
+//   }
+// }
```
Comment (kevintumbo) [inline src/components/Authentication/Signup/Home.js]: why do we have a whole file full of commented code?

**Review by kevintumbo — CHANGES_REQUESTED** [review]

File: src/components/Authentication/Signup/store.js (hunk 5 above)
Comment (Mutuba) [reply src/components/Authentication/Signup/store.js]: Sorry, the initial commit had that, I forgot to remove the store in the signup.  I have fixed that!

File: src/containers/Articles/Read/index.js (hunk 6 above)
Comment (Mutuba) [reply src/containers/Articles/Read/index.js]: I have done that!

File: src/store/store.js [hunk 12]
```diff
@@ -3,7 +3,6 @@ import { createStore, applyMiddleware } from 'redux';
 import { logger } from 'redux-logger';
 import thunk from 'redux-thunk';
 
-
 import rootReducer from './rootReducer';
 
 const middlewares = [logger, thunk];
```
Comment (kevintumbo) [inline src/store/store.js]: expose your initial state to the store.

File: src/styles/styles.scss (hunk 8 above)
[the remark this answers is not in the document]
Comment (kevintumbo) [reply src/styles/styles.scss:7]: properties not ordered yet.

File: src/styles/styles.scss (hunk 9 above)
[the remark this answers is not in the document]
Comment (kevintumbo) [reply src/styles/styles.scss]: this hasn't been addressed yet.

File: src/styles/styles.scss (hunk 10 above)
[the remark this answers is not in the document]
Comment (kevintumbo) [reply src/styles/styles.scss]: this hasn't been addressed yet.

**Review by kevintumbo — CHANGES_REQUESTED** [review]

**Review by kevintumbo — APPROVED** [review]

**Review by DerKip — APPROVED** [review]

**Review by Derrickkip — APPROVED** [review]

**Pull request merged**